pygn-mode.el

```diff
@@ -1322,20 +1313,49 @@ diagnostic tests were successful."
 With numeric prefix ARG, advance ARG games."
   (interactive "p")
   (cl-callf or arg 1)
-  (when (looking-at-p "\\[Event ")
-    (goto-char (line-end-position)))
-  (pygn-mode--next-game-driver arg))
+  (let ((node (pygn-mode--true-containing-node '(game series_of_games))))
+    (dotimes (_ arg)
+      (cond
+        ((eq 'series_of_games (tsc-node-type node))
+         (let ((newpos (save-excursion
+                         (skip-syntax-forward "-")
+                         (point))))
+           (if (pygn-mode--true-containing-node 'game newpos)
+               (progn
+                 (goto-char newpos)
+                 (setq node (pygn-mode--true-containing-node 'game)))
+             (error "No more games"))))
+        (t
+         (setq node (tsc-get-next-sibling node))
+         (if node
+             (goto-char (pygn-mode--true-node-first-position node))
+           (error "No more games"))))))
+  (pygn-mode-focus-game-at-point))
 
 (defun pygn-mode-previous-game (arg)
   "Move back to the previous game in a multi-game PGN buffer.
 
 With numeric prefix ARG, move back ARG games."
   (interactive "p")
   (cl-callf or arg 1)
-  (save-match-data
-    (unless (looking-at-p "\\[Event ")
-      (re-search-backward "^\\[Event " nil t))
-    (pygn-mode--next-game-driver (* arg -1))))
+  (let ((node (pygn-mode--true-containing-node '(game series_of_games))))
```

Comment:
Looks like there's a bit of code duplication. Do you think the code is made more or less manageable if it's consolidated?

Reply:
Less manageable here.  Though I would like to do a refactoring pass after many of these nuggets get merged.